account-kit/infra/src/middleware/gasManager.ts

```diff
@@ -432,6 +432,23 @@ const generateSignedPermit = async <
     ],
   });
 
+  const [decimals_response, nonce_response, paymasterData] = await Promise.all([
+    decimals_future,
+    nonce_future,
+    paymasterData_future,
+  ]);
+  if (!decimals_response.data) {
+    throw new Error("No decimals returned from erc20 contract call");
+  }
+  if (!nonce_response.data) {
+    throw new Error("No nonces returned from erc20 contract call");
+  }
+
+  const decimals =
+    10n ** (decimals_response.data ? BigInt(decimals_response.data) : 18n);
```

Comment:
`decimals_response.data` is always present because of the `if (!decimals_response.data)` check a few lines up.